classes/Visualizer/Module/Chart.php

```diff
@@ -1452,6 +1456,17 @@ public function getQueryData() {
 	public function saveQuery() {
 		check_ajax_referer( Visualizer_Plugin::ACTION_SAVE_DB_QUERY . Visualizer_Plugin::VERSION, 'security' );
 
+		if ( ! current_user_can( 'administrator' ) ) {
+			wp_send_json_error( array( 'msg' => __( 'Action not allowed for this user.', 'visualizer' ) ) );
+		}
+		if ( ! is_super_admin() ) {
+			wp_send_json_error( array( 'msg' => __( 'Action not allowed for this user.', 'visualizer' ) ) );
+		}
```

Comment:
NIT. Since they are related, you can group those into a single `if`. Something like:

```php
// User permissions
if (
    ! current_user_can( 'administrator' ) ||
    ! is_super_admin() 
) {
    wp_send_json_error( array( 'msg' => __( 'Action not allowed for this user.', 'visualizer' ) ) );
}
```